src/ctap2/attestation.rs

```diff
@@ -284,23 +290,47 @@ impl<'de> Deserialize<'de> for AuthenticatorData {
 }
 
 impl AuthenticatorData {
+    // see https://www.w3.org/TR/webauthn-2/#sctn-authenticator-data
+    // Authenticator Data
+    //                   Name Length (in bytes)
+    //               rpIdHash 32
+    //                  flags 1
+    //              signCount 4
+    // attestedCredentialData variable (if present)
+    //             extensions variable (if present)
     pub fn to_vec(&self) -> Result<Vec<u8>, AuthenticatorError> {
         let mut data = Vec::new();
-        data.extend(self.rp_id_hash.0);
-        data.extend([self.flags.bits()]);
-        data.extend(self.counter.to_be_bytes());
+        data.extend(self.rp_id_hash.0); // (1) "rpIDHash", len=32
+        data.extend([self.flags.bits()]); // (2) "flags", len=1 (u8)
+        data.extend(self.counter.to_be_bytes()); // (3) "signCount", len=4, 32-bit unsigned big-endian integer.
 
-        // TODO(baloo): need to yield credential_data and extensions, but that dependends on flags,
-        //              should we consider another type system?
+        // TODO(MS): Here flags=AT needs to be set, but this data comes from the security device
```

Comment:
Do we have bugs open for these todos?

Reply:
No, because I'm not even sure if there is anything "to do" here at all :) I can open one (or two), if you like.

Reply:
I'm not sure I understand the comment, so yes let's open a bug to discuss further.

Reply:
-> #233